Sources/SKCore/BuildSystemManager.swift

```diff
@@ -72,6 +72,22 @@ public actor BuildSystemManager {
   }
 }
 
+/// Splits `message` on newline characters such that each chunk is at most `maxChunkSize` bytes long.
+///
+/// The intended use case for this is to split compiler arguments into multiple chunks so that each chunk doesn't exceed
+/// the maximum message length of `os_log` and thus won't get truncated.
+private func splitLongMultilineMessage(message: String, maxChunkSize: Int) -> [String] {
+  var chunks: [String] = [""]
+  for line in message.split(separator: "\n", omittingEmptySubsequences: false) {
+    if chunks.last!.utf8.count + line.utf8.count < maxChunkSize {
+      chunks[chunks.count - 1] += "\n" + line
```

Comment:
Is it expected that the first chunk will start with a newline?

Reply:
Oh, no, that wasn’t intentional. Fixed it